client/src/App.js

```diff
@@ -16,14 +16,19 @@ import './App.css';
 function App() {
   const user = React.useContext(UserContext);
   const friends = React.useContext(FriendsContext);
+  const [userLoading, setUserLoading] = useState(false);
 
   React.useEffect(() => {
     const fetchData = async () => {
       if (!user._id) {
-        await user.getCurrent();
+        setUserLoading(true);
+        await user.getCurrent((err) => {
+          setUserLoading(false);
+        });
       }
 
       if (user._id) {
```

Comment:
I understand why you didn't use else here, because you want to run both blocks.
But you actually run both blocks with if, because the useEffect actually can run twice.

In your case this is what happens:
1- first it runs on mount and goes into the !user._id block
2- sets loading = true
2- awaits the user.getCurrent
3- checks if user._id again. But the problem is that the setState on the userContext is async, so the _id will still be undefined. The code works because as soon as the user is set, the useEffect code runs again (because it has [user]) and now it has an _id and can go to the second block.

So my suggestion would be to just use the regular if statement:
`
if (!user._id) {
...
} else {
...
}`